pkg/controller/kabaneroplatform/landing.go

```diff
@@ -70,27 +71,79 @@ func deployLandingPage(k *kabanerov1alpha1.Kabanero, c client.Client) error {
 		return err
 	}
 
-	// Create a clientset to drive API operations on resources.
-	config, err := clientcmd.BuildConfigFromFlags("", "")
+	// Retrieve the kabanero landing URL.
+	landingURL, err := getLandingURL(k, c)
 	if err != nil {
 		return err
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
+	// Create a Deployment. The landing application requires knowledge of the landingURL
+	// post route creation.
+	f, err = rev.OpenOrchestration("kabanero-landing-deployment.yaml")
 	if err != nil {
 		return err
 	}
 
-	// Retrieve the kabanero landing URL.
-	landingURL, err := getLandingURL(k, c, config)
+	s, err = renderOrchestration(f, templateContext)
 	if err != nil {
 		return err
 	}
 
-	// Create a Deployment. The landing application requires knowledge of the landingURL
-	// post route creation.
-	env := []corev1.EnvVar{{Name: "LANDING_URL", Value: landingURL}}
-	err = createDeployment(k, clientset, c, "kabanero-landing", image, env, nil, nil, kllog)
+	m, err = mf.FromReader(strings.NewReader(s), c)
+	if err != nil {
+		return err
+	}
+
+	transforms = []mf.Transformer{
+		mf.InjectOwner(k),
+		mf.InjectNamespace(k.GetNamespace()),
+		kabTransforms.AddEnvVariable("LANDING_URL", landingURL),
+	}
+
+	// See if we should define the OAuth volume and variables
+	secretInstance := &corev1.Secret{}
+	secretName := "kabanero-github-oauth-secret"
+	err = c.Get(context.Background(), types.NamespacedName{
+		Name:      secretName,
+		Namespace: k.ObjectMeta.Namespace}, secretInstance)
+
+	if err == nil {
+		// The secret exists, go ahead and put in the oauth information
+		transforms = append(transforms, kabTransforms.MountSecret(secretName, "/etc/oauth"))
+
+		// Try and figure out what the OAuth URL for github is, based on the API url.  If the API
+		// URL is not set, assume that it is "api.github.com".
+		apiUrlString := k.Spec.Github.ApiUrl
```

Comment:
Are the properties in `k.Spec.Github.ApiUrl` case insensitive? Our docs show `spec` `github` and `apiurl` as all lowercase when configuring a CR instance.

Reply:
The golang structs for the mapping of the CRD always uppercase the first letter of each property, because that's how go differentiates public and private things.  If the first letter is upper case, it's public, and if not, it's private.